### ISSUE TITLE
"--print-rules" throws SystemError

### ISSUE THREAD

**Issue by xiaohan2012** [mrapp-ke/Boomer#6]
Hi,

My environment:

- CentOS Linux release 7.9.2009 (Core)
- Python 3.8.7
- mlrl-boomer==0.7.1
- virtualenv

When I use Boomer in its command line mode, e.g.,  if I run:

`boomer --data-dir datasets/MULAN/bibtex --dataset bibtex --store-rules True`

things go well.

However, if I add `--print-rules`, e.g., `boomer --data-dir datasets/MULAN/bibtex --dataset bibtex --store-rules True --print-rules true`

The following error is given:

```
Traceback (most recent call last):
  File "model.pyx", line 430, in mlrl.common.cython.model.RuleModelVisitorWrapper.__visit_partial_head
  File "stringsource", line 251, in View.MemoryView.array_cwrapper
SystemError: <class 'mlrl.common.cython.model.array'> returned a result with an error set

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "model.pyx", line 414, in mlrl.common.cython.model.RuleModelVisitorWrapper.__visit_conjunctive_body
  File "stringsource", line 251, in View.MemoryView.array_cwrapper
SystemError: <class 'mlrl.common.cython.model.array'> returned a result with an error set

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "model.pyx", line 430, in mlrl.common.cython.model.RuleModelVisitorWrapper.__visit_partial_head
  File "stringsource", line 251, in View.MemoryView.array_cwrapper
SystemError: <class 'mlrl.common.cython.model.array'> returned a result with an error set

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "model.pyx", line 414, in mlrl.common.cython.model.RuleModelVisitorWrapper.__visit_conjunctive_body
  File "stringsource", line 251, in View.MemoryView.array_cwrapper
SystemError: <class 'mlrl.common.cython.model.array'> returned a result with an error set

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "model.pyx", line 430, in mlrl.common.cython.model.RuleModelVisitorWrapper.__visit_partial_head
  File "stringsource", line 251, in View.MemoryView.array_cwrapper
SystemError: <class 'mlrl.common.cython.model.array'> returned a result with an error set

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "model.pyx", line 414, in mlrl.common.cython.model.RuleModelVisitorWrapper.__visit_conjunctive_body
  File "stringsource", line 251, in View.MemoryView.array_cwrapper
SystemError: <class 'mlrl.common.cython.model.array'> returned a result with an error set

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "model.pyx", line 430, in mlrl.common.cython.model.RuleModelVisitorWrapper.__visit_partial_head
  File "stringsource", line 251, in View.MemoryView.array_cwrapper
SystemError: <class 'mlrl.common.cython.model.array'> returned a result with an error set

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "model.pyx", line 414, in mlrl.common.cython.model.RuleModelVisitorWrapper.__visit_conjunctive_body
  File "stringsource", line 251, in View.MemoryView.array_cwrapper
SystemError: <class 'mlrl.common.cython.model.array'> returned a result with an error set

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "model.pyx", line 430, in mlrl.common.cython.model.RuleModelVisitorWrapper.__visit_partial_head
  File "stringsource", line 251, in View.MemoryView.array_cwrapper
SystemError: <class 'mlrl.common.cython.model.array'> returned a result with an error set

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "model.pyx", line 414, in mlrl.common.cython.model.RuleModelVisitorWrapper.__visit_conjunctive_body
  File "stringsource", line 251, in View.MemoryView.array_cwrapper
SystemError: <class 'mlrl.common.cython.model.array'> returned a result with an error set

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "model.pyx", line 430, in mlrl.common.cython.model.RuleModelVisitorWrapper.__visit_partial_head
  File "stringsource", line 251, in View.MemoryView.array_cwrapper
SystemError: <class 'mlrl.common.cython.model.array'> returned a result with an error set

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "model.pyx", line 414, in mlrl.common.cython.model.RuleModelVisitorWrapper.__visit_conjunctive_body
  File "stringsource", line 251, in View.MemoryView.array_cwrapper
SystemError: <class 'mlrl.common.cython.model.array'> returned a result with an error set

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "model.pyx", line 430, in mlrl.common.cython.model.RuleModelVisitorWrapper.__visit_partial_head
  File "stringsource", line 251, in View.MemoryView.array_cwrapper
SystemError: <class 'mlrl.common.cython.model.array'> returned a result with an error set

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "model.pyx", line 414, in mlrl.common.cython.model.RuleModelVisitorWrapper.__visit_conjunctive_body
  File "stringsource", line 251, in View.MemoryView.array_cwrapper
SystemError: <class 'mlrl.common.cython.model.array'> returned a result with an error set

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "model.pyx", line 430, in mlrl.common.cython.model.RuleModelVisitorWrapper.__visit_partial_head
  File "stringsource", line 251, in View.MemoryView.array_cwrapper
SystemError: <class 'mlrl.common.cython.model.array'> returned a result with an error set

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "model.pyx", line 414, in mlrl.common.cython.model.RuleModelVisitorWrapper.__visit_conjunctive_body
  File "stringsource", line 251, in View.MemoryView.array_cwrapper
SystemError: <class 'mlrl.common.cython.model.array'> returned a result with an error set

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "model.pyx", line 430, in mlrl.common.cython.model.RuleModelVisitorWrapper.__visit_partial_head
  File "stringsource", line 251, in View.MemoryView.array_cwrapper
SystemError: <class 'mlrl.common.cython.model.array'> returned a result with an error set

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "model.pyx", line 414, in mlrl.common.cython.model.RuleModelVisitorWrapper.__visit_conjunctive_body
  File "stringsource", line 251, in View.MemoryView.array_cwrapper
SystemError: <class 'mlrl.common.cython.model.array'> returned a result with an error set

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "model.pyx", line 430, in mlrl.common.cython.model.RuleModelVisitorWrapper.__visit_partial_head
  File "stringsource", line 251, in View.MemoryView.array_cwrapper
SystemError: <class 'mlrl.common.cython.model.array'> returned a result with an error set

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "model.pyx", line 414, in mlrl.common.cython.model.RuleModelVisitorWrapper.__visit_conjunctive_body
  File "stringsource", line 251, in View.MemoryView.array_cwrapper
SystemError: <class 'mlrl.common.cython.model.array'> returned a result with an error set

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "model.pyx", line 430, in mlrl.common.cython.model.RuleModelVisitorWrapper.__visit_partial_head
  File "stringsource", line 251, in View.MemoryView.array_cwrapper
SystemError: <class 'mlrl.common.cython.model.array'> returned a result with an error set

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "model.pyx", line 414, in mlrl.common.cython.model.RuleModelVisitorWrapper.__visit_conjunctive_body
  File "stringsource", line 251, in View.MemoryView.array_cwrapper
SystemError: <class 'mlrl.common.cython.model.array'> returned a result with an error set

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "model.pyx", line 430, in mlrl.common.cython.model.RuleModelVisitorWrapper.__visit_partial_head
  File "stringsource", line 251, in View.MemoryView.array_cwrapper
SystemError: <class 'mlrl.common.cython.model.array'> returned a result with an error set

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "model.pyx", line 414, in mlrl.common.cython.model.RuleModelVisitorWrapper.__visit_conjunctive_body
  File "stringsource", line 251, in View.MemoryView.array_cwrapper
SystemError: <class 'mlrl.common.cython.model.array'> returned a result with an error set

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "model.pyx", line 430, in mlrl.common.cython.model.RuleModelVisitorWrapper.__visit_partial_head
  File "stringsource", line 251, in View.MemoryView.array_cwrapper
SystemError: <class 'mlrl.common.cython.model.array'> returned a result with an error set

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/scratch/cs/mldb/han/interpretable-mlc-code/.venv/bin/boomer", line 8, in <module>
    sys.exit(main())
  File "/scratch/cs/mldb/han/interpretable-mlc-code/.venv/lib/python3.8/site-packages/mlrl/testbed/main_boomer.py", line 131, in main
    runnable.run(parser)
  File "/scratch/cs/mldb/han/interpretable-mlc-code/.venv/lib/python3.8/site-packages/mlrl/testbed/runnables.py", line 44, in run
    self._run(args)
  File "/scratch/cs/mldb/han/interpretable-mlc-code/.venv/lib/python3.8/site-packages/mlrl/testbed/runnables.py", line 123, in _run
    experiment.run()
  File "/scratch/cs/mldb/han/interpretable-mlc-code/.venv/lib/python3.8/site-packages/mlrl/testbed/experiments.py", line 62, in run
    super().run()
  File "/scratch/cs/mldb/han/interpretable-mlc-code/.venv/lib/python3.8/site-packages/mlrl/testbed/training.py", line 62, in run
    self.__train_test_split()
  File "/scratch/cs/mldb/han/interpretable-mlc-code/.venv/lib/python3.8/site-packages/mlrl/testbed/training.py", line 158, in __train_test_split
    self._train_and_evaluate(meta_data, None, train_x, train_y, None, test_x, test_y, first_fold=0,
  File "/scratch/cs/mldb/han/interpretable-mlc-code/.venv/lib/python3.8/site-packages/mlrl/testbed/experiments.py", line 142, in _train_and_evaluate
    model_printer.print(learner_name, meta_data, current_learner, current_fold=current_fold,
  File "/scratch/cs/mldb/han/interpretable-mlc-code/.venv/lib/python3.8/site-packages/mlrl/testbed/model_characteristics.py", line 204, in print
    text = self._format_model(meta_data, model)
  File "/scratch/cs/mldb/han/interpretable-mlc-code/.venv/lib/python3.8/site-packages/mlrl/testbed/model_characteristics.py", line 270, in _format_model
    model.visit(formatter)
SystemError: <method 'visit' of 'mlrl.common.cython.model.RuleModel' objects> returned a result with an error set
```

**Comment by mrapp-ke** [comment]
Thank you for reporting this issue. The `--print-rules true` argument is currently broken if a data set contains nominal attributes. As a workaround you can use the `--store-rules true` option or add the following argument `--print-options '{print_nominal_values=false}'`. The problem will be fixed in the next release.

**Comment by mrapp-ke** [comment]
Should be fixed in the latest version 0.8.2.
